[sphinx] Adds Logging section in Developer Guide

matteobarbera opened the pull request:
I've written a page to detail some of the logging options that are available for Paparazzi. Not everything that I saw on the Paparazzi Wiki made it here because from my understanding most of the other options will soon become deprecated. 

I wasn't 100% sure about the information on the Pprzlog format so comments on that section are welcome.

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+are printing information simultaneously.
+
+When running in simulation ``fprintf`` automatically prints the message in the Paparazzi Center. You can also
+read these messages on the real drone using the following steps:
```
I suggest to specify that this example is only valid for Bebop 1 et 2.

matteobarbera replied:
Is this not possible with every Linux based OS that has WiFi and specifies a host for a telnet connection? E.g. the Disco, although during flight you wouldn't be able to maintain WiFi connection

gautierhattenberger replied:
It's true that it works for Disco which is sharing the same hardware than Bebop more or less. The only Linux-based system that we support are Parrot drones, so I guess it is fine even if there are so things specific to these platforms, like default IP or directory tree.

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+.. code-block:: xml
+
+  <firmware name="rotorcraft">
+    <module name="tlsf"/>
+    <module name="pprzlog"/>
+    <module name="logger" type="sd_chibios"/>
+    <module name="flight_recorder"/>
```
With the new dependency solver, it is only necessary to load `flight_recorder` module.

Also, when using `sd_chibios` only, a log file is opened and can be used to log any type of data in ASCII (or even binary if needed) like a regular file. When loading the flight recorder module, two separated files are opened. Have a look at the `meteo_stick` module for an example of using the different options depending of the module's configuration.

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+
+.. code-block:: bash
+
+  ~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG
```
`~/paparazzi/sw/logalizer/sd2log fr_XXXX.LOG output_dir_path` to extract the log in an other folder than default log path

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+
+- A ``.log`` file, an XML file, which contains a copy of the whole configuration (airframes, flight plans, ...)
+- A ``.data`` file, an ascii file, which contains the list of the received messages. Each message is time-stamped in
+  milliseconds since the creation of the file and marked with the ID of the sending aircraft
```
actually a floating point value in seconds with 3 digits

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+- A ``.log`` file, an XML file, which contains a copy of the whole configuration (airframes, flight plans, ...)
+- A ``.data`` file, an ascii file, which contains the list of the received messages. Each message is time-stamped in
+  milliseconds since the creation of the file and marked with the ID of the sending aircraft
+- A ``.tlm`` file, ???
```
the `tlm` file is a copy of the original `.LOG` file, with the same naming `data_time.*`. It makes it easier to associate with decoded log files.
By the way, a recommended use is to directly call `sd2log` on the file still on the SD card mounted as a mass storage, it is not really needed to copy it first on the computer.

gautierhattenberger commented on doc/sphinx/source/developer_guide/logging.rst:
```diff
@@ -0,0 +1,264 @@
+
+In this case, at the time the message was logged, the attitude of the drone corresponded to :math:`{\phi} = 0.036228`,
+:math:`{\theta} = 0.018550`, and :math:`{\psi} = 0.021443` radians. Note that the appropriate ``messages.xml``
+description, i.e. the one which has been used while th elog was created, is itself stored in the associated ``.log``
```
`the log`

matteobarbera commented:
I've addressed the changes (except for the Bebop 1/2 comment, I thought it worked with every Linux based OS that has WiFi and that specifies a host) and added a section on SD ChibiOS logger covering how to create custom log functions for binary and ASCII logs. 

I used `meteo_stick.c` as a reference as suggested, but as I've never attempted to do something like this it's very possible that something is incorrect in my explanation  :sweat_smile:
